streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -213,6 +212,31 @@ public <K, V> void send(final String topic,
                     keyClass,
                     valueClass),
                 exception);
+        } catch (final SerializationException exception) {
+            final ProducerRecord<K, V> record = new ProducerRecord<>(topic, partition, timestamp, key, value, headers);
+            final ProductionExceptionHandler.ProductionExceptionHandlerResponse response;
+            try {
+                response = productionExceptionHandler.onSerializationException(record, exception);
+            } catch (final Exception e) {
+                log.error("Fatal handling serialization exception on record {}", record, e);
+                recordSendError(topic, e, null);
+                return;
+            }
+
+            if (response == ProductionExceptionHandlerResponse.FAIL) {
+                log.error("Fatal handling serialization exception on record {}", record, exception);
+                recordSendError(topic, exception, null);
+                return;
+            }
+
+            log.warn("Unable to serialize {}. Continue processing. " +
+                    "ProducerRecord(key=[{}], value=[{}], topic=[{}], partition=[{}], timestamp=[{}])",
+                keyBytes == null ? "key" : "value",
```

Comment:
Even if `keyBytes` is `null`, it's not clear if key or value failed -- the key could have been `null`.

Reply:
hmm good point, maybe just log the continue processing message then.